internal/printer/printer_test.go

```diff
@@ -1595,6 +1595,34 @@ import { Container, Col, Row } from 'react-bootstrap';
 				code: `${$$maybeRenderHead($$result)}<div class:list="something"${$$spreadAttributes(Astro.props,"Astro.props")}></div>`,
 			},
 		},
+		{
+			name:   "class list",
+			source: `<div class:list={['one', 'variable']} />`,
+			want: want{
+				code: `${$$maybeRenderHead($$result)}<div${$$addAttribute(['one', 'variable'], "class:list")}></div>`,
+			},
+		},
+		{
+			name:   "class and class list simple array",
+			source: `<div class="two" class:list={['one', 'variable']} />`,
+			want: want{
+				code: `${$$maybeRenderHead($$result)}<div${$$addAttribute(['two', ['one', 'variable']], "class:list")}></div>`,
+			},
+		},
+		{
+			name:   "class and class list object",
+			source: `<div class="two three" class:list={['hello goodbye', { hello: true, world: true }]} />`,
+			want: want{
+				code: `${$$maybeRenderHead($$result)}<div${$$addAttribute(['two three', ['hello goodbye', { hello: true, world: true }]], "class:list")}></div>`,
+			},
+		},
+		{
+			name:   "class and class list set",
+			source: `<div class="two three" class:list={[ new Set([hello: true, world: true]) ]} />`,
+			want: want{
+				code: `${$$maybeRenderHead($$result)}<div${$$addAttribute(['two three', [ new Set([hello: true, world: true]) ]], "class:list")}></div>`,
```

Comment:
hmm how was that passing tests??

Reply:
The compiler parser the content of the class list as text, I think. I don't think the compiler parser the content against valid JavaScript, otherwise we would need a whole JavaScript parsing

Reply:
wacky.